src/Workspaces/Core/Portable/Workspace/Solution/SolutionState_Checksum.cs

```diff
@@ -156,10 +157,22 @@ public async Task<Checksum> GetChecksumAsync(ProjectId projectId, CancellationTo
                     var analyzerReferenceChecksums = ChecksumCache.GetOrCreate<ChecksumCollection>(AnalyzerReferences,
                         _ => new ChecksumCollection(AnalyzerReferences.SelectAsArray(r => serializer.CreateChecksum(r, cancellationToken))));
 
-                    var projectChecksums = await Task.WhenAll(projectChecksumTasks).ConfigureAwait(false);
+                    var allResults = await Task.WhenAll(projectChecksumTasks).ConfigureAwait(false);
+                    using var _1 = ArrayBuilder<ProjectId>.GetInstance(projectChecksumTasks.Length, out var projectIds);
```

Comment:
>projectChecksumTasks [](http://example.com/codeflow?start=71&length=20)

tiny nit: iterate over allResults below, so might be more clear to allocate based on it's Length

Reply:
will do!